ducktape/tests/runner.py

```diff
@@ -113,6 +113,12 @@ def run_all_tests(self):
                 if not self.session_context.no_teardown:
                     self.log(logging.INFO, "tearing down")
                     self.teardown_single_test()
+                else:
```

Comment:
Would it make more sense to just add a flag to `teardown_single_test` so we keep the code unified and don't duplicate parts of it? I'm actually noticing that we also don't free nodes (which is fine here since reusing them doesn't make much sense if you wanted to preserve the running services) and that with this copied code, we also don't set the `stop_testing` flag if something goes wrong.

Neither is critical now since both should be fine with the no teardown flag since they only affect things when you run multiple tests, but I'm wondering if we're just going to keep having to think about the two code paths separately and remember to fill things in for both.